addons/sale_timesheet/models/project.py

```diff
@@ -537,7 +537,7 @@ def _get_default_partner_id(self, project, parent):
                 return related_project.sale_line_employee_ids.sale_line_id.order_partner_id[:1]
         return res
 
-    sale_order_id = fields.Many2one(domain="['|', '|', ('partner_id', '=', partner_id), ('partner_id', 'child_of', commercial_partner_id), ('partner_id', 'parent_of', partner_id)]")
+    sale_order_id = fields.Many2one(domain="['|', '|', ('partner_id', '=', partner_id), ('partner_id.commercial_partner_id.id', 'parent_of', partner_id), ('partner_id', 'parent_of', partner_id)]")
```

Comment:
`.id` is really needed? If yes, why?

Does it not equivalent to:

```suggestion
    sale_order_id = fields.Many2one(domain="['|', '|', ('partner_id', '=', partner_id), ('partner_id.commercial_partner_id', 'parent_of', partner_id), ('partner_id', 'parent_of', partner_id)]")
```
? 🤔

Reply:
Wow, I did not know that 🤯 

Thanks for the explanation! 🙂 

Then LGTM! 💪🏼